src/com/dabomstew/pkrandom/cli/CliRandomizer.java

```diff
@@ -0,0 +1,188 @@
+                new Gen6RomHandler.Factory(),
+                new Gen7RomHandler.Factory()
+        };
+
+        Settings settings;
+        try {
+            File fh = new File(settingsFilePath);
+            FileInputStream fis = new FileInputStream(fh);
+            settings = Settings.read(fis);
+            // taken from com.dabomstew.pkrandom.newgui.NewRandomizerGUI.saveROM, set distinctly from all other settings
+            settings.setCustomNames(FileFunctions.getCustomNames());
+            fis.close();
+        } catch (UnsupportedOperationException ex) {
+            ex.printStackTrace();
+            System.err.println(ex.getMessage());
+            return false;
+        } catch (IllegalArgumentException ex) {
+            ex.printStackTrace();
+            System.err.println(bundle.getString("GUI.invalidSettingsFile"));
+            return false;
+        } catch (IOException ex) {
+            ex.printStackTrace();
+            System.err.println(bundle.getString("GUI.settingsLoadFailed"));
+            return false;
+        }
+
+        try {
+            File romFileHandler = new File(sourceRomFilePath);
+            RomHandler romHandler;
+
+            for (RomHandler.Factory rhf : checkHandlers) {
+                if (rhf.isLoadable(romFileHandler.getAbsolutePath())) {
+                    romHandler = rhf.create(RandomSource.instance());
+                    romHandler.loadRom(romFileHandler.getAbsolutePath());
+                    if (saveAsDirectory && romHandler.generationOfPokemon() != 6 && romHandler.generationOfPokemon() != 7) {
+                        saveAsDirectory = false;
+                        System.err.println("WARNING: Saving as directory does not make sense for non-3DS games, ignoring \"-d\" flag...");
+                    }
+
+                    CliRandomizer.displaySettingsWarnings(settings, romHandler);
+
+                    if (!saveAsDirectory) {
+                        List<String> extensions = new ArrayList<>(Arrays.asList("sgb", "gbc", "gba", "nds", "cxi"));
+                        extensions.remove(romHandler.getDefaultExtension());
+
+                        File fh = new File(destinationRomFilePath);
+                        fh = FileFunctions.fixFilename(fh, romHandler.getDefaultExtension(), extensions);
+                        if (romHandler instanceof AbstractDSRomHandler || romHandler instanceof Abstract3DSRomHandler) {
+                            String currentFN = romHandler.loadedFilename();
+                            if (currentFN.equals(fh.getAbsolutePath())) {
+                                System.err.println(bundle.getString("GUI.cantOverwriteDS"));
+                                return false;
+                            }
+                        }
+                    }
+
+
+                    Randomizer randomizer = new Randomizer(settings, romHandler, saveAsDirectory);
+                    randomizer.randomize(destinationRomFilePath);
```

Comment:
My bad, the change I suggested for `fixFileName` doesn't quite make everything work. In order for the file name fix to have any effect, `randomize` has to use the path stored in the file handler returned by `fixFileName`. (The same thing is done in `NewRandomizerGUI`, [see here.](https://github.com/Ajarmar/universal-pokemon-randomizer-zx/blob/master/src/com/dabomstew/pkrandom/newgui/NewRandomizerGUI.java#L774)) With that in mind, `File fh = new File(destinationRomFilePath);` should be put before the `if (!saveAsDirectory)` clause.
```suggestion
                    randomizer.randomize(fh.getAbsolutePath());
```